akka-serialization-jackson/src/test/scala/akka/serialization/jackson/JacksonSerializerSpec.scala

```diff
@@ -96,6 +97,15 @@ object ScalaTestMessages {
 
   final case class OldCommandNotInBindings(name: String)
 
+  object Planet extends Enumeration {
+    type Planet = Value
+    val Mercury, Venus, Earth, Mars, Krypton = Value
+  }
+  final case class Alien(name:String, planet:Planet.Planet) extends TestMessage
+
+  class PlanetType extends TypeReference[Planet.type] {}
+  final case class Superhero(name:String, @JsonScalaEnumeration(classOf[PlanetType]) planet:Planet.Planet) extends TestMessage
```

Comment:
is it possible to enable the string value format by default, using class names in the format should be avoided

Reply:
It's not possible in Jackson 2.2 but `It is expected this behavior will be deprecated in version 2.3, and removed entirely in 3.0.`( [ref](https://github.com/FasterXML/jackson-module-scala/wiki/Enumerations)). The annotation is used to provide the `class PlanetType extends TypeReference[Planet.type]` information that's missing when not adding the classname in the serialized form.

It'd be great if a setting enabled a compile or runtime check to ensure only the single-String format (based on the metadata provided by `@JsonScalaEnumeration`) is used.

I guess we could add a small mention in the docs?

Reply:
yes, we have a section Annotations where we could mention such gotcha
https://doc.akka.io/docs/akka/current/serialization-jackson.html#annotations